homeassistant/components/livisi/binary_sensor.py

```diff
@@ -0,0 +1,149 @@
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DOMAIN,
+    LIVISI_REACHABILITY_CHANGE,
+    LIVISI_STATE_CHANGE,
+    LOGGER,
+    WDS_DEVICE_TYPE,
+)
+from .coordinator import LivisiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up binary_sensor device."""
+    coordinator: LivisiDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    @callback
+    def handle_coordinator_update() -> None:
+        """Add Window Sensor."""
+        shc_devices: list[dict[str, Any]] = coordinator.data
+        entities: list[BinarySensorEntity] = []
+        for device in shc_devices:
+            if (
+                device["type"] == WDS_DEVICE_TYPE
+                and device["id"] not in coordinator.devices
+            ):
+                livisi_binary: BinarySensorEntity = create_entity(
+                    config_entry, device, coordinator
+                )
+                LOGGER.debug("Include device type: %s", device["type"])
+                coordinator.devices.add(device["id"])
+                entities.append(livisi_binary)
+        async_add_entities(entities)
+
+    config_entry.async_on_unload(
+        coordinator.async_add_listener(handle_coordinator_update)
+    )
+
+
+def create_entity(
+    config_entry: ConfigEntry,
+    device: dict[str, Any],
+    coordinator: LivisiDataUpdateCoordinator,
+) -> BinarySensorEntity:
+    """Create Binary Sensor Entity."""
+    config_details: dict[str, Any] = device["config"]
+    capabilities: list = device["capabilities"]
+    room_id: str = device["location"]
+    room_name: str = coordinator.rooms[room_id]
+    livisi_binary = LivisiWindowSensor(
+        config_entry,
+        coordinator,
+        unique_id=f"{device['id']}_window_sensor_state",
+        manufacturer=device["manufacturer"],
+        device_type=device["type"],
+        name=config_details["name"],
+        capability_id=capabilities[0],
+        room=room_name,
+    )
+    return livisi_binary
+
+
+class LivisiWindowSensor(
+    CoordinatorEntity[LivisiDataUpdateCoordinator], BinarySensorEntity
+):
+    """Represents the Livisi Binary Sensor."""
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        coordinator: LivisiDataUpdateCoordinator,
+        unique_id: str,
+        manufacturer: str,
+        device_type: str,
+        name: str,
+        capability_id: str,
+        room: str,
+    ) -> None:
+        """Initialize the Livisi Binary."""
+        self.config_entry = config_entry
+        self._attr_unique_id = unique_id
+        self._attr_name = name
+        self._attr_device_class = BinarySensorDeviceClass.WINDOW
```

Comment:
Not all Livisi Sensors are connected to a window. device["tags"]["typeCategory"] may be either "TCWindowId" or "TCDoorId", so the class could also be set to "BinarySensorDeviceClass.DOOR"